_RFCs/002-node-scaffolding-split.md

```diff
@@ -0,0 +1,60 @@
+
+# Motivation
+[motivation]: #motivation
+
+There are currently two ways of running Node applications. Some applications (especially those created with the [Express](https://expressjs.com/) framework) run the application using Node.js's web server. Others - especially those built with [React](https://reactjs.org/) and [Angular](https://angular.io/) - compile Javascript, etc. as static assets and run them behind a proxy like Nginx. Supporting both of these approaches through one scaffolding would be deeply complex at best and untenable at worst.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This splits the node scaffolding into two different scaffoldings - one for dynamic applications and one for applications that are compiled to static assets.
+
+If you are a node developer creating an Express application (or any other application that uses the Node.js web server) you should use the scaffolding for dynamic applications.
+
+If you are a node developer creating a React or Angular application (or any other application that is compiled into static assets, then run behind a proxy like Nginx) you should use the scaffolding for static applications.
+
+Developers who are using the current node scaffolding need to migrate to either the dynamic scaffolding or the static scaffolding. The dynamic scaffolding will largely be the same as the current node scaffolding, but users will need to change the name of the scaffolding they specify when initializing habitat in their application repos (and in their habitat/plan.sh files).
+
+The original scaffolding-node will remain on builder but will be deprecated and no longer updated.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Scaffolding-node splits into two scaffoldings - scaffolding-node-dynamic and scaffolding-node-static (these are just placeholder names, they can be something else).
+
+Scaffolding-node-dynamic remains largely the same as the current scaffolding-node.
+
+Scaffolding-node-static will compile a node application into static assets and will run them behind an Nginx proxy (there will be a dependency on core/nginx).
+
+The code around hab plan init would need to be changed to accept either the scaffolding-node-dynamic or scaffolding-node-static scaffoldings as an -s argument.
+
+All tutorials/documentation that refer to scaffolding-node will need to change to refer to either scaffolding-node-dynamic or scaffolding-node-static.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The original intent of the node scaffolding was to be a quickstart for Habitizing any node application. This approach would force node developers to think about which scaffolding would serve them best and choose one or the other.
```

Comment:
Another drawback would be having to support two scaffoldings that could diverge, but probably shouldn't. What about using a single scaffolding for which we expose configuration? You'll probably need to do a couple of things in the case of a static app:

```
pkg_scaffolding="core/scaffolding-node"

do_build() {
  npm run build  # Since these these will vary by project
}
```

... whereas in a server app, you might do something like:

```
pkg_scaffolding="core/scaffolding-node"
pkg_svc_run="npm start"  # Since these will vary, too 

do_build() {
  npm run build-prod-or-something
}
```

The implication here would be like, if a scaffolding is provided and `pkg_svc_run` is provided, then set the dep as make run hook with it; otherwise, only set a build dep. Given that these npm commands will vary in ways we can't predict, something like this could give us the flexibility we need without burdening the user with too much to think/learn about.

Reply:
I am open to this - it might not be as quick of a start, but would provide much more flexibility as the node ecosystem evolves.  This also is in line with one of Habitat's original focuses - "just enough configuration."